hbase-client/src/main/java/org/apache/hadoop/hbase/client/RawAsyncHBaseAdmin.java

```diff
@@ -758,16 +758,13 @@ public CompletableFuture<Void> disableTable(TableName tableName) {
    * targetState).
    */
   private static CompletableFuture<Boolean> completeCheckTableState(
-    CompletableFuture<Boolean> future, TableState tableState, Throwable error,
+    CompletableFuture<Boolean> future, Optional<TableState> tableState, Throwable error,
```

Comment:
I don't think that this change solves the source of the NPE. Per the stack trace, `tableState` was `null` entering the lambda on line 797. I was expecting to find a null-check or use of `Optional.ofNullable` in order to ride over the `null` value.

I think that to solve the stack trace, you need to guard against a `null` value being provided here in `completeCheckTableState` for the `tableState` parameter. You could use a basic `if (tableState == null)`, or an `Optional` chain that unpacks the value, something like:

```
Optional.ofNullable(tableState).flatMap(Function.identity).ifPresentOrElse(...)
```

I agree that passing the `Optional` all the way down is a nicer approach.

Reply:
We should never return null when the return value is Optional, otherwise why not just returning the object directly without wrapping Optonal?
The only exception is that an exception is thrown, and this is what we want to fix here. In the old code, we use the returned Optional object before checking error, so if error is not null, we will get a NPE.
In this PR, we pass the Optional down to the completeCheckTableState method, and only check it after we make sure error is null.